DOCKER.md

```diff
@@ -0,0 +1,123 @@
+# Development with Docker
```

Comment:
Not very important, and a distinct element of bikeshedding, but I'd advocate putting this in SETUP.md instead of creating another top-level .md file.

Reply:
(100% ignorable if you disagree btw, I would not want to block forward progress here on something silly and very debatable)